util/topgen.py

```diff
@@ -562,23 +562,26 @@ def generate_otp_ctrl(topcfg: Dict[str, object], cfg_path: Path, out_path: Path,
 
 # generate ac_range_check with ipgen
 def generate_ac_range_check(topcfg: Dict[str, object], out_path: Path) -> None:
-    # Not all tops have an ac range check instance
-    if 'ac_range_check' not in topcfg:
-        return
-
     log.info('Generating ac_range_check with ipgen')
     topname = topcfg['name']
 
+    # Determine RACL params from the top-level config, otherwise use ipgen's default values
+    racl_params = {}
+    if 'racl_config' in topcfg:
+        racl_params = {
+            'nr_role_bits': topcfg['racl']['nr_role_bits'],
+            'nr_ctn_uid_bits': topcfg['racl']['nr_ctn_uid_bits']
+        }
+
     # Get the AC Range Check instance
     ac_ranges = lib.find_module(topcfg['module'], 'ac_range_check')
     params = {
-        "nr_role_bits": 4,
-        "nr_ctn_uid_bits": 5,
-        "num_ranges": topcfg['ac_range_check']['num_ranges'],
-        "module_instance_name": ac_ranges['type']
+        'num_ranges': ac_ranges['ipgen_param']['num_ranges'],
+        'module_instance_name': ac_ranges['type']
     }
+    params.update(racl_params)
 
-    ipgen_render("ac_range_check", topname, params, out_path)
+    ipgen_render('ac_range_check', topname, params, out_path)
```

Comment:
Can we not have quote vs double quotes discrepancies? The overwhelming number of strings are double quoted, and we have the policy of preserving the prevailing convention. You can see topcfg/validate.py is mostly single quotes.